lib/rules/curly.js

```diff
@@ -309,6 +309,98 @@ module.exports = {
                 hasUnsafeIf(statement) && isFollowedByElseKeyword(node);
         }
 
+        /**
+         * Examples:
+         *
+         * The following location should be reported
+         *
+         * if (foo) bar()
+         *         ^
+         *
+         * while(foo) bar();
+         *           ^
+         *
+         * for(;;) bar();
+         *         ^
+         *
+         * for (var foo of bar) console.log(foo)
+         *                      ^
+         *
+         * for (var foo in bar) console.log(foo)
+         *                      ^
+         *
+         * Get the location of missing parent location where it should have been placed.
+         * @param {ASTNode} node the node to check.
+         * @param {string} name name to check.
+         * @returns {Object} Position object.
+         */
+        function getMissingLoc(node, name) {
+            switch (name) {
+                case "else":
+                    return getElseKeyword(node).loc.start;
+                case "if":
+                    return astUtils.getNextLocation(sourceCode, node.test.loc.end);
+                case "while":
+                    return astUtils.getNextLocation(sourceCode, node.test.loc.end);
+                case "for":
+                    return node.body.loc.start;
+                case "for-of":
+                    return node.body.loc.start;
+                case "do":
+                    return node.body.loc.start;
+                case "for-in":
+                    return node.body.loc.start;
+                default:
+                    return node.loc;
+            }
+        }
+
+        /**
+         * Examples:
+         *
+         * The following location should be reported
+         *
+         * if (true) foo(); else { baz(); }
+         *                       ^
+         *
+         * for (;;) { foo(); }
+         *          ^
+         *
+         * while (bar) { foo(); }
+         *             ^
+         *
+         * do{foo();} while(bar);
+         *   ^
+         *
+         * for (var foo of bar) {console.log(foo)}
+         *                      ^
+         *
+         * Get the location of unexpected parens. mostly the starting parens `{`
+         * @param {ASTNode} node the node to check.
+         * @param {string} name name to check.
+         * @returns {Object} Position object.
+         */
+        function getUnexpectedParenLoc(node, name) {
+            switch (name) {
+                case "else":
+                    return getElseKeyword(node).loc.start;
```

Comment:
@mdjermanovic Do we want to include `loc.end` as well for these cases?

Reply:
I'd say - yes. If we are targeting the start location of a particular token, then it probably makes sense to pass the end location of that token as well.

Reply:
for unexpected curly braces, I think the start location can be fine as its a single token so start and end location will be the same. 

for missing curly brace,  i will make the changes. 

let me know if this is fine ?

Reply:
> for unexpected curly braces, I think the start location can be fine as its a single token so start and end location will be the same.

For a single-character token, the end location isn't the same as its start location. The end location would be that with `column + 1`

Reply:
ok then I think it will show like this 

```js
if(foo) {bar}
 //     ^^
```

right ?

Reply:
> ok then I think it will show like this
> 
> ```js
> if(foo) {bar}
>  //     ^^
> ```

It would underline just `{`, because the character at the end loc isn't considered as being included.

It's like a selection between cursor that is positioned at `start` and cursor that is positioned at `end`.